client/src/App.tsx

```diff
@@ -94,15 +94,30 @@ const App = () => {
   const [roots, setRoots] = useState<Root[]>([]);
   const [env, setEnv] = useState<Record<string, string>>({});
 
+  // Check for runtime config injected by the server
+  const runtimeConfig = window.__RUNTIME_CONFIG__ || {};
+
   const [config, setConfig] = useState<InspectorConfig>(() => {
     const savedConfig = localStorage.getItem(CONFIG_LOCAL_STORAGE_KEY);
-    if (savedConfig) {
-      return {
-        ...DEFAULT_INSPECTOR_CONFIG,
-        ...JSON.parse(savedConfig),
-      } as InspectorConfig;
+    let configFromStorage = savedConfig
+      ? ({
+          ...DEFAULT_INSPECTOR_CONFIG,
+          ...JSON.parse(savedConfig),
+        } as InspectorConfig)
+      : DEFAULT_INSPECTOR_CONFIG;
+
+    // Override with runtime injected values if available
+    if (runtimeConfig.MCP_PROXY_FULL_ADDRESS) {
```

Comment:
Shouldn't this be more like:

```typescript    
    if (runtimeConfig.MCP_PROXY_FULL_ADDRESS) {
      configFromStorage = {
        ...configFromStorage,
        MCP_PROXY_FULL_ADDRESS: {
          value: runtimeConfig.MCP_PROXY_FULL_ADDRESS,
        },
      };
```

Reply:
are there properties other than `value` on `MCP_PROXY_FULL_ADDRESS` that are important?

Reply:
It is a [ConfigItem](https://github.com/modelcontextprotocol/inspector/blob/main/client/src/lib/configurationTypes.ts#L1).